specification/search/data-plane/Microsoft.Azure.Search.Data/preview/2017-11-11-preview/searchindex.json

```diff
@@ -224,10 +543,88 @@
           }
         }
       }
-      
     }
   },
   "definitions": {
+    "HitHighlights": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string"
+        }
+      },
+      "description": "Defines the generic document in an Azure Search index.",
+      "x-ms-external": true
```

Comment:
I don't think this needs to be custom. It's just a dictionary of string to list of string, which you can model with `additionalProperties` (see examples in searchservice.json).